### PR TITLE
FEAT: Hawaii county

### PR DESCRIPTION
**Pull request by tlyon3**
Can get the data from the dashboard fine, but I can't get the county level specific data. It shows up on the dashboard when you select a specific county.

Tried finding a filter function name, but the user actions don't have a linkSpec field that has the filter function name.

If anyone else wants to take a look, they are more than welcome.

### CODE REVIEW

**Comment by sglyon** [comment]
cc @christopherturner if  you happen to have any spare cycles to look at this and maybe find a filter name, that would be helpful!

File: can_tools/scrapers/official/HI/hi_county.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+from can_tools.scrapers.official.base import (
+    TableauDashboard,
+    TableauMapClick,
+    StateDashboard,
+)
+
+from can_tools.scrapers.util import requests_retry_session
+
+
+class HawaiiVaccineCounty(TableauDashboard, StateDashboard):
```
Comment (christopherturner) [inline can_tools/scrapers/official/HI/hi_county.py:17]: TableauDashboard is a subclass of StateDashboard, so this line can be:

```python
class HawaiiVaccineCounty(TableauDashboard):
```

File: can_tools/scrapers/official/HI/hi_county.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    source = "https://health.hawaii.gov/coronavirusdisease2019/what-you-should-know/current-situation-in-hawaii/#vaccine"
+    source_name = (
+        "State of Hawai'i - Department of Health"
+        "Disease Outbreak Control Division | COVID-19"
+    )
+    location_type = "county"
+    state_fips = int(us.states.lookup("Hawaii").fips)
+    # https://public.tableau.com/shared/H33ZZ9HCC?:display_count=y&:origin=viz_share_link&:embed=y
+    # Hawai'i https://public.tableau.com/shared/7TCBHC568?:display_count=y&:origin=viz_share_link&:embed=y
+    filterFunctionName = f"county%20select" # this is the name of the user action, not the filter function name. doesn't work
```
Comment (christopherturner) [inline can_tools/scrapers/official/HI/hi_county.py:28]: The filter function is `[sqlproxy.0td6cgz0bpiy7x131qvze0jvbqr1].[none:County:nk]`, so this line should be:

```python
filterFunctionName = "[sqlproxy.0td6cgz0bpiy7x131qvze0jvbqr1].[none:County:nk]"
```

File: can_tools/scrapers/official/HI/hi_county.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    baseurl = "https://public.tableau.com"
+    viewPath = "HawaiiCOVID-19-VaccinationDashboard/VACCINESBYCOUNTY"
+    # baseurl = "https://public.tableau.com/shared/"
+    # viewPath = "7TCBHC568"
+
+    
+    def fetch(self):
+        # Get Maui County
+        self.filterFunctionValue = 'MAUI'
```
Comment (christopherturner) [inline can_tools/scrapers/official/HI/hi_county.py:37]: You'll want to use the following values (case-sensitive) for the `self.filterFunctionValue`:

- `Hawaii`
- `Honolulu`
- `Kauai`
- `Maui`

**Review by christopherturner — CHANGES_REQUESTED** [review]
Hey!  I was able to find the filter function name, and your code works great once you make that small adjustment!  You'll also want to use the case-sensitive values for the counties.  Happy to help if there's any questions I can answer!

File: can_tools/scrapers/official/HI/hi_county.py (hunk 2 above)
Comment (ghop02) [reply can_tools/scrapers/official/HI/hi_county.py:28]: Do we have a function that could make it easy to show all of the filters? I think it kinda exists on the TableauClickClass, but would be really handy to be able to see a report of that easily

File: can_tools/scrapers/official/HI/hi_county.py (hunk 3 above)
Comment (tlyon3) [reply can_tools/scrapers/official/HI/hi_county.py]: Done

File: can_tools/scrapers/official/HI/hi_county.py (hunk 2 above)
Comment (tlyon3) [reply can_tools/scrapers/official/HI/hi_county.py]: Used the `filterFunctionName`

File: can_tools/scrapers/official/HI/hi_county.py (hunk 1 above)
Comment (tlyon3) [reply can_tools/scrapers/official/HI/hi_county.py]: Removed `StateDashboard`

**Review by christopherturner — APPROVED** [review]

**Comment by sglyon** [comment]
@tlyon3 is this ready for review now?

**Comment by sglyon** [comment]
ahh I see that @christopherturner  reviewed, excellent!

I'll merge away

**Comment by christopherturner** [comment]
Hey @sglyon!  My apologies, I was trying to close out the changes I'd suggested regarding the filter function – I think I marked it as approved in the process.  Sorry for the confusion, I'll leave those suggestions open in the future so it doesn't reflect as approved prematurely.  I believe that further changes are required before `hi_county.py` is fully functional.

**Comment by christopherturner** [comment]
The code (with those modifications) did work for me when I ran through it locally yesterday.  However today, I'm experiencing issues with loading the Tableau dashboard, and I seem to be getting the same error in the browser.

**Comment by sglyon** [comment]
Oh shoot. Haha my fault. 

Ok we'll have to get this one fixed up

Do you have any bandwidth to take a look at it?

**Comment by tlyon3** [comment]
> The code (with those modifications) did work for me when I ran through it locally yesterday. However today, I'm experiencing issues with loading the Tableau dashboard, and I seem to be getting the same error in the browser.
> 
> When I enter the tableau dashboard URL by hand, I'm getting static images instead of HTML pages. And, when I visit the [source](https://health.hawaii.gov/coronavirusdisease2019/what-you-should-know/current-situation-in-hawaii/#vaccine), it isn't loading on my device.

I'm getting the same results here to...maybe this is a problem on their end?

**Pull request merged**